libcnb-data/src/buildpack/stack.rs

```diff
@@ -48,3 +48,69 @@ pub enum StackError {
     #[error("Invalid Stack ID: {0}")]
     InvalidStackId(#[from] StackIdError),
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn deserialize_specific_stack_without_mixins() {
+        let toml_str = r#"
+id = "heroku-20"
+"#;
+        assert_eq!(
+            toml::from_str::<Stack>(toml_str),
+            Ok(Stack::Specific {
+                // Cannot use the `stack_id!` macro due to: https://github.com/Malax/libcnb.rs/issues/179
+                id: "heroku-20".parse().unwrap(),
+                mixins: Vec::new()
+            }),
+        );
+    }
+
+    #[test]
+    fn deserialize_specific_stack_with_mixins() {
+        let toml_str = r#"
+id = "io.buildpacks.stacks.focal"
+mixins = ["build:jq", "wget"]
+"#;
+        assert_eq!(
+            toml::from_str::<Stack>(toml_str),
+            Ok(Stack::Specific {
+                id: "io.buildpacks.stacks.focal".parse().unwrap(),
+                mixins: vec![String::from("build:jq"), String::from("wget")]
+            }),
+        );
+    }
+
+    #[test]
+    fn deserialize_any_stack() {
+        let toml_str = r#"
+id = "*"
+"#;
+        assert_eq!(toml::from_str::<Stack>(toml_str), Ok(Stack::Any));
+    }
+
+    #[test]
+    fn reject_specific_stack_with_invalid_name() {
+        let toml_str = r#"
+id = "io.buildpacks.stacks.*"
+"#;
+        let err = toml::from_str::<Stack>(toml_str).unwrap_err();
+        assert!(err
```

Comment:
For the failure cases, I've intentionally used an intermediate variable (unlike the success cases), since in the event the test fails with a different error message, it means one can more easily insert additional statements here before the assert to debug further (for those not using the debugger).